### PR TITLE
Add treemap code and pretty-print for bytes

### PR DESCRIPTION
**Pull request by peterdjlee**
Originally reviewed and approved in #2084 - breaking it into a separate PR to land only the Treemap code and code related to pretty-printing bytes

### CODE REVIEW

File: packages/devtools_app/test/support/utils.dart [hunk 1]
```diff
@@ -42,6 +44,46 @@ Future<Timeline> loadNetworkProfileTimeline() async {
...
+  final processedTestData = treemapFromJson(treemapTestData);
+  processedTestData['n'] = 'Root';
+
+  return generateTree(processedTestData);
+}
+
+// TODO(peterdjlee): Remove this method once code size PR (#2107) lands.
+/// Builds a tree with [TreemapNode] from [treeJson] which represents
+/// the hierarchical structure of the tree.
+TreemapNode generateTree(Map<String, dynamic> treeJson) {
```
Comment (peterdjlee) [inline packages/devtools_app/test/support/utils.dart:61]: This code also lives in `CodeSizeController` which I was planning on using after deleting this method once the code size PR (#2107) lands. Instead, should I be taking this method out to something like `treemap_utils.dart`?

File: packages/devtools_app/test/treemap_test.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+void main() {
+  group('Treemap', () {
+    TreemapNode root;
+
+    setUp(() async {
+      root = await loadInstructionSizesJson();
+    });
+
+    void changeRoot(TreemapNode newRoot) {
```
Comment (peterdjlee) [inline packages/devtools_app/test/treemap_test.dart:20]: @kenzieschmoll This is also a duplicate that exists in `CodeSizeController`. Should I be taking it out to a different file or is it okay if I keep both?

File: packages/devtools_app/test/support/utils.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/support/utils.dart:61]: yeah. let's add here and figure out how to clean up in your code size cl

File: packages/devtools_app/test/treemap_test.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/treemap_test.dart:20]: i don't think you actually need this method. just pass a null or noop callback ( `() {}` ) for onRootChangedCallback

Comment (peterdjlee) [reply packages/devtools_app/test/treemap_test.dart:23]: I think if we want to test what happens after zooming in we need the function

**Pull request merged**